src/musicbrainz/web/musicbrainzrecordingstask.cpp

```diff
@@ -161,11 +177,31 @@ void MusicBrainzRecordingsTask::doNetworkReplyFinished(
         auto trackReleases = m_trackReleases.values();
         m_trackReleases.clear();
         emitSucceeded(trackReleases);
-        return;
+        return true;
     }
 
     // Continue with next recording id
     DEBUG_ASSERT(!m_queuedRecordingIds.isEmpty());
+    emit currentRecordingFetched();
+    auto inhibitTimerElapsed = m_measurementTimer.elapsed(true);
+    qDebug() << "Task took:" << inhibitTimerElapsed.toIntegerMillis();
+    qDebug() << kMinTimeBetweenMbRequests;
+    m_requestTimer.setSingleShot(true);
+    if (inhibitTimerElapsed.toIntegerMillis() >= kMinTimeBetweenMbRequests) {
+        qDebug() << "Task took more than a second, slot is calling now.";
+        m_measurementTimer.restart(true);
+        m_requestTimer.start(1);
```

Comment:
I think start() implies restart()

Reply:
Yeah you are right, [Documentation](https://doc.qt.io/qt-6/qtimer.html#start) says that `If the timer is already running, it will be stopped and restarted.` 

Thank you.

Reply:
Sorry, I think we need that restart because it restarts the Timer of ours, not the QTimer. Without the restart, the `m_measurementTimer` stacks up and all the requests starts right away.